app/services/authentication/providers/apple.rb

```diff
@@ -0,0 +1,88 @@
+module Authentication
+  module Providers
+    # Apple authentication provider, uses omniauth-apple as backend
+    class Apple < Provider
+      OFFICIAL_NAME = "Apple".freeze
+      SETTINGS_URL = "https://appleid.apple.com/account/manage".freeze
+      TRUSTED_CALLBACK_ORIGIN = "https://appleid.apple.com".freeze
+      CALLBACK_PATH = "/users/auth/apple/callback".freeze
+
+      def new_user_data
+        # Apple sends `first_name` and `last_name` as separate fields
+        name = "#{info.first_name} #{info.last_name}"
+        timestamp = raw_info.id_info.auth_time
+
+        user_data = {
+          email: info.email,
+          apple_created_at: Time.zone.at(timestamp),
+          apple_username: user_nickname,
+          name: name
+        }
+
+        if Rails.env.test?
+          user_data[:profile_image] = SiteConfig.mascot_image_url
```

Comment:
**Question**: Is there a way to do this without specially adding test code to production?

Reply:
Great question! The reason why this was added is because directly setting `remote_profile_image_url` triggered CarrierWave to fetch the image, which caused the tests to fail for me (the original PR had something similar in place).

I'm not sure if there's a way we can just leave the else clause, i.e. `user_data[:remote_profile_image_url] = Users::ProfileImageGenerator.call`

If anyone can share an idea on how to solve this, like a way to mock CarrierWave somehow (not as trivial as it sounds at least for me) I'd be happy to include it!

This is all behind a feature flag, so we can brainstorm and improve on this when making it generally available (getting rid of the flag) too 👍

Reply:
Cool: I think this definitely works for now and I'd hate to hold up the PR because of it!